syntax = "proto3";

package service;

message Document {
    repeated float vector_data = 1;
    string id = 2;
    map<string, string> attributes = 3;
}

message SearchRequest {
    repeated float vector_data = 1;
    optional int32 k = 2;
}

message SearchResponse {
    repeated string ids  = 1;
}

service BaseService {
    rpc ProcessSearchRequest (SearchRequest) returns (SearchResponse) {}
}